proxied-token/proxied-token-spec.ini

```diff
@@ -0,0 +1,295 @@
+[root]
+k: #execute => #halt
+code: {PROXY_CODE}
+
+; This are our variables:
+; Default statusCode is EVMC_REVERT
+statusCode: _ => EVMC_REVERT
+output: _ => _
+callData:
+log: _
+refund: _ => _
+msg_sender: RANDOM_ID
+callValue: 0
+storage:
+    #hashedLocation({COMPILER}, {_MASTER_COPY}, .IntList) |-> MASTER_COPY_ID
+requires:
+
+; Proxy pattern functions
+
+[implementation]
+callData: #abiCallData("implementation", .TypedArgs)
+
+[implementation-success]
+statusCode: _ => EVMC_SUCCESS
+output: _ => #asByteStackInWidth(MASTER_COPY_ID, 32)
+
+[implementation-failure]
+; non-zero call value
+callValue: CALL_VALUE
++requires:
+    andBool #rangeUInt(256, CALL_VALUE)
+    andBool 0  <Int CALL_VALUE
+
+[proxyType]
+callData: #abiCallData("proxyType", .TypedArgs)
+
+[proxyType-success]
+statusCode: _ => EVMC_SUCCESS
+output: _ => #asByteStackInWidth(2, 32)
+
+[proxyType-failure]
+; non-zero call value
+callValue: CALL_VALUE
++requires:
+    andBool #rangeUInt(256, CALL_VALUE)
+    andBool 0  <Int CALL_VALUE
+
+[changeMasterCopy]
+callData: #abiCallData("changeMasterCopy", #address(NEW_MASTER_COPY_ID))
++storage:
+    #hashedLocation({COMPILER}, {_CREATOR}, .IntList) |-> CREATOR_ID
++requires:
+    andBool #rangeAddress(NEW_MASTER_COPY_ID)
+
+[changeMasterCopy-success]
+statusCode: _ => EVMC_SUCCESS
+msg_sender: CREATOR_ID
+storage:
+    #hashedLocation({COMPILER}, {_MASTER_COPY}, .IntList) |-> (MASTER_COPY_ID => NEW_MASTER_COPY_ID)
+    #hashedLocation({COMPILER}, {_CREATOR}, .IntList) |-> CREATOR_ID
++requires:
+    andBool 0  <Int  NEW_MASTER_COPY_ID 
+
+[changeMasterCopy-failure]
```

Comment:
empty intermediate specs like this have no effect. Please delete.

Reply:
done